src/main/java/org/kuali/kfs/krad/service/impl/RemoteModuleServiceBase.java

```diff
@@ -0,0 +1,653 @@
+        if (this.kualiConfigurationService == null) {
+            this.kualiConfigurationService = KRADServiceLocator.getKualiConfigurationService();
+        }
+
+        return this.kualiConfigurationService;
+    }
+
+    public void setKualiConfigurationService(ConfigurationService kualiConfigurationService) {
+        this.kualiConfigurationService = kualiConfigurationService;
+    }
+
+    /**
+     * @see org.springframework.context.ApplicationContextAware#setApplicationContext(org.springframework.context.ApplicationContext)
+     */
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    /**
+     * This overridden method ...
+     *
+     * @see ModuleService#listAlternatePrimaryKeyFieldNames(java.lang.Class)
+     */
+    @Override
+    public List<List<String>> listAlternatePrimaryKeyFieldNames(Class businessObjectInterfaceClass) {
+        return null;
+    }
+
+    /**
+     * This method determines whether or not this module is currently locked
+     *
+     * @see ModuleService#isLocked()
+     */
+    @Override
+    public boolean isLocked() {
+        LOG.debug("isLocked, entering Cornell customization");
+        boolean shouldLock = false;
+        ModuleConfiguration configuration = this.getModuleConfiguration();
+        if (configuration != null) {
+            String namespaceCode = configuration.getNamespaceCode();
+            String componentCode = KRADConstants.DetailTypes.ALL_DETAIL_TYPE;
+            String parameterName = KRADConstants.SystemGroupParameterNames.OLTP_LOCKOUT_ACTIVE_IND;
+            ParameterService parameterService = CoreFrameworkServiceLocator.getParameterService();
+            String shouldLockout = parameterService.getParameterValueAsString(namespaceCode, componentCode, parameterName);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("isLocked, nameSpace: " + namespaceCode + " componentCode: " + componentCode + " parameterName: " + parameterName);
+                LOG.debug("isLocked, shouldLockout: " + shouldLockout);
+            }
+            if (StringUtils.isNotBlank(shouldLockout)) {
+                shouldLock = parameterService.getParameterValueAsBoolean(namespaceCode, componentCode, parameterName) && !isUserInLockoutOverrideList();
+            }
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("isLocked, shouldLock: " + shouldLock);
+        }
+        return shouldLock;
+    }
+    
+    protected boolean isUserInLockoutOverrideList() {
+        String[] overrideList = StringUtils.split(getOverrideUsersList(), ",");
+        String loggedInUserPrincipleName = GlobalVariables.getUserSession().getPerson().getPrincipalName();
+        Object[] filteredList = Arrays.stream(overrideList).filter(x -> x.equalsIgnoreCase(loggedInUserPrincipleName)).toArray();
```

Comment:
I think it might make more sense to use a parameter evaluator to check if the username is in the list. The list might need to be separated by ";" in that case..